### PR TITLE
[SYCL][NFC] Revise Cmake config for LIT

### PR DESCRIPTION
**Pull request by vladimirlaz**
CMake target configuration is aligned with LIT test location.

That also alows to run on-device LIT tests for prebuilt SYCL compiler
and runtime binaries following the
[instructions](https://github.com/intel/llvm/blob/sycl/sycl/test/on-device/README.md).

### CODE REVIEW

**Review by pvchupin — DISMISSED** [review]
Looks OK to me. Would be good to have review from others.

File: sycl/test/on-device/README.md [hunk 1]
```diff
@@ -11,3 +11,59 @@ It is developer responsibility to move the tests from this directory to
...
+# clone llvm_project repo
+git clone https://github.com/intel/llvm sycl_lit
+cd sycl_lit
+export ROOT=`pwd`
+mkdir build
+cd build
+
+# set extra environment
+export EXTCMPLRROOT=# path to deployed SYCL compiler and runtime
+export GET_DEVICE_TOOL=# Path to get full path to utility detecting available devices ([source](../../tools/get_device_count_by_type.cpp))
```
Comment (romanovvlad) [inline sycl/test/on-device/README.md:31]:
```suggestion
export GET_DEVICE_TOOL=# Path to get_device_count_by_type tool ([source](../../tools/get_device_count_by_type.cpp))
```
I think the link(source) can be confusing here, one might think that they need to specify path to the source file.

File: sycl/test/on-device/README.md [hunk 2]
```diff
@@ -11,3 +11,59 @@ It is developer responsibility to move the tests from this directory to
...
+cmake -DLLVM_TARGETS_TO_BUILD="X86" -DLLVM_EXTERNAL_PROJECTS=sycl-test \
+      -DLLVM_EXTERNAL_SYCL_TEST_SOURCE_DIR=$ROOT/sycl/test/on-device \
+      -DSYCL_SOURCE_DIR=$ROOT/sycl -DOpenCL_LIBRARIES=$EXTCMPLRROOT/lib \
+      $ROOT/llvm
+
+# Build LIT tools
+make FileCheck not llvm-config
+
+# Run tests for OpenCL BE
+/usr/bin/python3.6 $ROOT/llvm/utils/lit/lit.py -v \
```
Comment (romanovvlad) [inline sycl/test/on-device/README.md:44]: Is it necessary to hardcode path to the python and it's version?

File: sycl/test/on-device/lit.cfg.py [hunk 3]
```diff
@@ -81,7 +81,7 @@
 config.substitutions.append( ('%BE_RUN_PLACEHOLDER', "env SYCL_DEVICE_FILTER={SYCL_PLUGIN} ".format(SYCL_PLUGIN=backend)) )
 config.substitutions.append( ('%RUN_ON_HOST', "env SYCL_DEVICE_FILTER=host ") )
 
-get_device_count_by_type_path = os.path.join(config.llvm_tools_dir, "get_device_count_by_type")
+get_device_count_by_type_path = lit_config.params.get('GET_DEVICE_TOOL', os.path.join(config.llvm_tools_dir, "get_device_count_by_type"))
```
Comment (romanovvlad) [inline sycl/test/on-device/lit.cfg.py:84]: Can we add some diagnostic message which says if param is set and which tool is finally going to be used?

File: sycl/test/on-device/README.md (hunk 2 above)
Comment (vladimirlaz) [reply sycl/test/on-device/README.md:44]: fixed

**Comment by pvchupin** [comment]
Good comments @romanovvlad, thank you!

File: sycl/test/on-device/README.md (hunk 1 above)
Comment (vladimirlaz) [reply sycl/test/on-device/README.md:31]: fixed

File: sycl/test/on-device/lit.cfg.py (hunk 3 above)
Comment (vladimirlaz) [reply sycl/test/on-device/lit.cfg.py:84]: fixed

**Review by romanovvlad — APPROVED** [review]
LGTM

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Pull request merged**